test/xpu_api/numerics/complex.number/cases.h

```diff
@@ -218,48 +214,45 @@ CLANG_DIAGNOSTIC_IGNORED_AUTOLOGICAL_CONSTANT_COMPARE
         return non_zero_nan;
     }
     return non_zero;
-
-CLANG_DIAGNOSTIC_POP
 }
 
 inline
 int
 classify(double x)
 {
-// Suppress clang warning: comparison with infinity always evaluates to false in fast floating point modes [-Wtautological-constant-compare]
-CLANG_DIAGNOSTIC_PUSH
-CLANG_DIAGNOSTIC_IGNORED_AUTOLOGICAL_CONSTANT_COMPARE
-
     if (x == 0)
         return zero;
     if (std::isinf(x))
         return inf;
     if (std::isnan(x))
         return NaN;
     return non_zero;
-
-CLANG_DIAGNOSTIC_POP
 }
 
 void is_about(float x, float y)
 {
-// Suppress clang warning: floating-point comparison is always true; constant cannot be represented exactly in type 'float' [-Wliteral-range]
-CLANG_DIAGNOSTIC_PUSH
-CLANG_DIAGNOSTIC_IGNORED_LITERAL_RANGE
-
-    assert(std::abs((x-y)/(x+y)) < 1.e-6);
-
-CLANG_DIAGNOSTIC_POP
+    if (x != y)
```

Comment:
We add this check because nothing required to calculate in case when ```x``` exactly is the same as ```y```